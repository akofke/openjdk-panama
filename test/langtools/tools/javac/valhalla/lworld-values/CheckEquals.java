/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support == or !=
 *
 * @compile/fail/ref=CheckEquals.out -XDrawDiagnostics CheckEquals.java
 */

final value class CheckEquals {
    boolean foo(CheckEquals a, CheckEquals b) {
        return (a == b) || (a != b);
    }
    int x = 10;
}
