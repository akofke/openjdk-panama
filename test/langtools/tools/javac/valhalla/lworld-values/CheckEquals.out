CheckEquals.java:10:19: compiler.err.value.does.not.support: ==
CheckEquals.java:10:31: compiler.err.value.does.not.support: !=
2 errors
